rs/sns/governance/src/governance.rs

```diff
@@ -5079,55 +5079,70 @@ impl Governance {
         true
     }
 
-    /// Refreshes the cached_upgrade_steps field
-    pub async fn refresh_cached_upgrade_steps(&mut self) {
-        let Some(deployed_version) = self.proto.deployed_version.as_ref() else {
-            log!(
-                ERROR,
-                "Cannot refresh cached_upgrade_steps: deployed_version not set."
-            );
-            return;
+    pub fn try_temporarily_lock_refresh_cached_upgrade_steps(&mut self) -> Result<Version, String> {
+        let deployed_version = self
+            .proto
+            .deployed_version
+            .clone()
+            .ok_or("Cannot lock refresh_cached_upgrade_steps: deployed_version not set.")?;
+
+        let mut cached_upgrade_steps = match self.proto.cached_upgrade_steps_or_err() {
+            Ok(cached_upgrade_steps) => cached_upgrade_steps,
+            Err(message) => {
+                self.push_to_upgrade_journal(upgrade_journal_entry::UpgradeStepsReset::new(
+                    message,
+                    vec![deployed_version.clone()],
+                ));
+                CachedUpgradeSteps::without_pending_upgrades(
+                    deployed_version.clone(),
+                    self.env.now(),
+                )
+            }
         };
```

Comment:
Could you use cached_upgrade_steps = get_or_reset_cached_upgrade_steps? Or no, because that returns a clone when you need a mutable reference? I feel like this is duplicated code somehow

Reply:
Thanks for the suggestion, I think it's better to use `get_or_reset_cached_upgrade_steps` here indeed. 

Note that this will now also reset the target version, which seems fine.